### PR TITLE
Add adventure custom post type

### LINKED ISSUE #7: Declare and set up a custom post type for adventure travel results
**As an editor or author of a site running the Travel theme, I should have a way to built custom content for "adventure" travel results.**
AC1. A custom post type should be declared.
AC2. Some basic output (`the_title`, `the_content`, etc. or just whatever's inside `single`) should be built so that single views work for this post type.

### PR DESCRIPTION
**Pull request by DavidCramer**
Adds the `adventure` custom post type.

Fixes #7

### CODE REVIEW

**Comment by DavidCramer** [comment]
Still requires the additional metafields setup to capture the extra data, but otherwise as a basis for the custom post type definition, this is ready for review.

cc @ThierryA @westonruter

File: includes/class-amp-travel-cpt.php [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+		$args = array(
+			'labels'              => $labels,
+			'description'         => __( 'Adventure Custom Post Type for travel theme.', 'travel' ),
+			'public'              => true,
+			'exclude_from_search' => true,
+			'publicly_queryable'  => true,
+			'show_ui'             => true,
+			'show_in_nav_menus'   => true,
+			'show_in_menu'        => true,
+			'show_in_admin_bar'   => true,
```
Comment (westonruter) [inline includes/class-amp-travel-cpt.php]: This is missing `show_in_rest` which is required for Gutenberg.

File: includes/class-amp-travel-cpt.php [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+	 */
+	public function setup() {
+		/*
+		 * Enable support for Post Thumbnails on posts and pages.
+		 *
+		 * @link https://developer.wordpress.org/themes/functionality/featured-images-post-thumbnails/
+		 */
+		add_theme_support( 'post-thumbnails' );
+
+		// Image Sizes.
```
Comment (westonruter) [inline includes/class-amp-travel-cpt.php]: This should be a phpdoc comment to describe the filter.

**Review by westonruter — COMMENTED** [review]
For the meta fields, we'll want to use either Gutenberg blocks that use postmeta as their attribute sources.

**Comment by DavidCramer** [comment]
Thanks @westonruter - I have addressed the changes you requested. Feel free to review when you have time.

**Comment by miina** [comment]
@DavidCramer On the `adventure` post type -- did you also try editing in Gutenberg? When I create a new Adventure, publish it, and then change a setting (for example the content) and try to save again, I’m getting some errors + the title disappears from the view. Looks similar to behavior that got fixed [here](https://github.com/WordPress/gutenberg/pull/5738), however that was supposedly happening only if the CPT doesn't support `title`.

Wondering if that's something that only I'm experiencing or if it's an issue with Gutenberg.
screen shot 2018-04-09 at 2 48 48 pm

If you have a moment could you try if it works OK for you?

File: includes/class-amp-travel-cpt.php [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+ *
+ * @package WPAMPTheme
+ */
+
+/**
+ * Class AMP_Travel_CTP
+ *
+ * @package WPAMPTheme
+ */
+class AMP_Travel_CTP {
```
Comment (miina) [inline includes/class-amp-travel-cpt.php]: Looks like a typo, should probably be `_CPT` instead of `_CTP`.

Comment (DavidCramer) [reply includes/class-amp-travel-cpt.php]: oops yes, will fix

**Comment by DavidCramer** [comment]
@miina I haven't tested yet in Gutenberg yet as this was just the definition of the type. But I think I should put it back to WIP and get that aspect sorted.

**Comment by DavidCramer** [comment]
@westonruter - I initially did this just as the CPT definition. However, as @miina pointed out, it's not working yet with Gutenberg. I'll do more tests and clean it up first, then will let you know when you can review it again.

**Comment by DavidCramer** [comment]
@westonruter @ThierryA this is ready for review now. @miina I have tested with Gutenberg and it behaves as expected.

File: includes/class-amp-travel-cpt.php [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+			'items_list_navigation' => _x( 'Adventures list navigation', 'Screen reader text for the pagination heading on the post type listing screen. Default “Posts list navigation”/”Pages list navigation”.', 'travel' ),
+			'items_list'            => _x( 'Adventures list', 'Screen reader text for the items list heading on the post type listing screen. Default “Posts list”/”Pages list”.', 'travel' ),
+		);
+
+		$args = array(
+			'labels'                => $labels,
+			'description'           => __( 'Adventure Custom Post Type for travel theme.', 'travel' ),
+			'public'                => true,
+			'exclude_from_search'   => true,
+			'publicly_queryable'    => true,
```
Comment (kienstra) [inline includes/class-amp-travel-cpt.php]: The following arguments have the default value, and are not needed:
* publicly_queryable
* show_ui
* show_in_nav_menus
* show_in_menu
* show_in_admin_bar
* capability_type
* hierarchical
* query_var
* can_export
* rest_controller_class

File: includes/class-amp-travel-cpt.php [hunk 5]
```diff
@@ -0,0 +1,143 @@
...
+			'727x409',
+			'600x338',
+			'500x281',
+			'375x211',
+			'335x188',
+			'320x180',
+			'280x158',
+			'240x135',
+			'160x90',
+			'122x67',
```
Comment (kienstra) [inline includes/class-amp-travel-cpt.php]: Are all of these image sizes needed? 

You know more about the specifics than I do. But it seems excessive to create 15 image sizes for every image uploaded.

**Review by kienstra — CHANGES_REQUESTED** [review]
**Almost Ready, A Few Points**

Hi @DavidCramer,
Thanks for registering the new `'adventure'` post type. There are a few points here, one of which might not need any action.

File: includes/class-amp-travel-cpt.php (hunk 4 above)
Comment (DavidCramer) [reply includes/class-amp-travel-cpt.php]: @kienstra - yup, I'll remove them.

File: includes/class-amp-travel-cpt.php (hunk 5 above)
Comment (DavidCramer) [reply includes/class-amp-travel-cpt.php]: @kienstra - no idea. but it's a good point. I took these from the `ampconf` theme as a starting point. I'll reduce it down to what I think should be needed for the current blocks, etc.

Comment (kienstra) [reply includes/class-amp-travel-cpt.php]: This is something that we should revisit. But I'm going to merge this PR to unblock other PRs.

**Review by kienstra — APPROVED** [review]
**Approved**

If it's alright, I applied the 2 review points I made with [this commit](https://github.com/xwp/travel/pull/33/commits/5ee6fdc32c2dac3e28a4649973bba49d8b3cfe24). 

@DavidCramer, could you please look at whether there are any more image sizes needed here? I deleted the `add_image_size()` call for now, to merge this.

**Comment by DavidCramer** [comment]
Thanks @kienstra. Can always add additional sizes if needed later. It's all good.

**Pull request merged**